File: 700/718 - Maximum Length of Repeated Subarray.java
class Solution {
    public int findLength(int[] A, int[] B) {
        int len1 = A.length, len2 = B.length;
        int[][] dp = new int[len1 + 1][len2 + 1];
        int ret = 0;
        
        for (int i = 1; i <= len1; ++i) {
            for (int j = 1; j <= len2; ++j) {
                dp[i][j] = (A[i - 1] == B[j - 1] ? (dp[i - 1][j - 1] + 1) : 0);
                ret = Math.max(ret, dp[i][j]);
            }
        }
        
        return ret;
    }
}

File: csharp/718.maximum-length-of-repeated-subarray.cs
/*
 * @lc app=leetcode id=718 lang=csharp
 *
 * [718] Maximum Length of Repeated Subarray
 */

// @lc code=start
public class Solution {
    public int FindLength(int[] A, int[] B) {
        int m = A.Length, n = B.Length, ret = 0;
        int[,] dp = new int[m + 1, n + 1];
        for (int i = 1; i <= m; ++i) {
            for (int j = 1; j <= n; ++j) {
                if (A[i - 1] == B[j - 1])
                {
                    dp[i, j] = 1 + dp[i - 1, j - 1];
                    ret = Math.Max(ret, dp[i, j]);
                }
            }
        }
        return ret;
    }
}
// @lc code=end

